gui/wxpython/history/tree.py

```diff
@@ -0,0 +1,53 @@
+"""
+@package history_catalog::tree
+
+@brief History browser tree classes
+
+Classes:
+ - history_catalog::HistoryCatalogTree
+
+(C) 2023 by Linda Karlovska, and the GRASS Development Team
+
+This program is free software under the GNU General Public
+License (>=v2). Read the file COPYING that comes with GRASS
+for details.
+
+@author Linda Karlovska (Kladivova) linda.karlovska@seznam.cz
+"""
+
+import copy
+
+from core.treemodel import TreeModel, ModuleNode
+
+from grass.grassdb.history import read_history, get_current_mapset_gui_history_path
+
+
+class HistoryBrowserTree:
+    """Data class for the history browser tree of executed commands."""
+
+    def __init__(self, max_length=50):
+        self.model = TreeModel(ModuleNode)
+        self.max_length = max_length
+        self.CreateModel()
+
+    def _trim_text(self, text):
+        if len(text) > self.max_length:
+            return text[: self.max_length - 3] + "..."
+        else:
+            return text
```

Comment:
I don't see any benefit trimming the text, especially not with a fixed width. When you make the widget larger, you can't see the command. So I would just remove it, no?

Reply:
I agree, this solution is not general. The thing is that I do not like that commands are not wrapped by the edge of the pane. It is the same problem for the Modules tab.

I would like to have some functionality like "wrap commands based on pane width when changing the width" but it could possibly also slow down the loading of command texts in the pane..